### DISCUSSION TITLE
Make value optional from JRuleName

### DISCUSSION THREAD

**Discussion by querdenker2k** [seaside1/jrule#180] [Ideas]
I would like to make the value of the JRuleName annotation optional. But in generell i name it like the method name.
And one more but breaking step would be to rename the JRuleName annotation to JRuleMethod.
@seaside1 @seime What would you say?

**Comment by seime** [comment]
What problem does it solve? 

I personally use a human readable name for the rule and personally prefer to keep it as it is. However, even as a breaking change the mitigation for users is simple (search+replace).

**Reply by querdenker2k** [reply]
My current "problem" is that i name it like the method-name and therefore it's duplicated defined. It could be still a parameter but not required, so that anybody can use another name then the method name. I use elasticsearch/kibana to search my logs and use just the mdc rule name. If no value is defined the default could be the method name.

For the second thing to rename the annotation is that in my mind it's clearer to define JRuleMethod as a rule method than JRuleName. Name should just be the name parameter of JRuleMethod.

Both of them is just cosmetic.

**Reply by seaside1** [reply]
I also prefer to use JRuleName, I think it is more reable. If it's optional I don't object into the change.
